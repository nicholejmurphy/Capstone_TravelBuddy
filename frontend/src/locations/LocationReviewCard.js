import React from "react";

import Divider from "@material-ui/core/Divider";
import { makeStyles } from "@material-ui/core";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import Link from "@material-ui/core/Link";
import Grid from "@material-ui/core/Grid";

const useStyles = makeStyles((theme) => ({
  button: {
    marginTop: "15px",
    marginBottom: "15px",
  },
  title: {
    marginTop: "20px",
    marginBottom: "5px",
  },
  description: {
    marginTop: "10px",
  },
}));

/** Shows details about a location's review
 * - Recieves review from parent: LocationReviewList
 * - Has access links to trip advisor for more details
 */
function LocationReviewCard({ review }) {
  const classes = useStyles();

  return (
    <div>
      <Typography variant="h6" className={classes.title}>
        {review.title}
      </Typography>
      <Grid container justifyContent="flex-start">
        <Grid item xs={12} align="left">
          <Typography variant="subtitle2">
            Date: {review.travel_date}
          </Typography>
        </Grid>
        <Grid item xs={6}>
          <Typography variant="subtitle2" component="p">
            Rating: {review.rating}{" "}
            <img alt="rating icon" src={review.rating_image_url} />
          </Typography>
        </Grid>
        <Grid item xs={6} align="right">
          <Typography variant="subtitle2">
            Trip type: {review.trip_type}
          </Typography>
        </Grid>
      </Grid>
      <Typography variant="body1" className={classes.description}>
        {review.text}
      </Typography>
      <Button
        component={Link}
        color="primary"
        href={review.web_url}
        target="_blank"
        className={classes.button}
        size="small"
      >
        Read more
      </Button>
      <Divider className={classes.divider} />
    </div>
  );
}

export default LocationReviewCard;
