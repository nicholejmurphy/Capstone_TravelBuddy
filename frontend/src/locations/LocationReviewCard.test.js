import React from "react";
import ReactDOM from "react-dom";
import { render } from "@testing-library/react";
import LocationReviewCard from "./LocationReviewCard";
import { MemoryRouter } from "react-router";

// smoke test
it("renders without crashing", function () {
  const div = document.createElement("div");
  ReactDOM.render(
    <MemoryRouter>
      <LocationReviewCard review={{}} />
    </MemoryRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

// snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <LocationReviewCard review={{}} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
