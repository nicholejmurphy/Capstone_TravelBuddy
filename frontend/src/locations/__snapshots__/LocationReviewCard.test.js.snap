// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <div>
      <p />
      <p>
        Date: 
      </p>
      <p>
        Rating:  
        <img
          alt="rating icon"
        />
      </p>
      <p>
        Trip type: 
      </p>
      <p />
      <small>
        <a>
          Read More on TripAdvisor
        </a>
      </small>
    </div>
  </div>
</DocumentFragment>
`;
